webapp/src/pluginConstants/apiService.ts

```diff
@@ -0,0 +1,56 @@
+// Plugin api service (RTK query) configs
+export const pluginApiServiceConfigs: Record<PluginApiServiceName, PluginApiService> = {
+    createTask: {
+        path: '/tasks',
+        method: 'POST',
+        apiServiceName: 'createTask',
+    },
+    createLink: {
+        path: '/link',
+        method: 'POST',
+        apiServiceName: 'createLink',
+    },
+    getAllLinkedProjectsList: {
+        path: '/project/link',
+        method: 'GET',
+        apiServiceName: 'getAllLinkedProjectsList',
+    },
+    unlinkProject: {
+        path: '/project/unlink',
+        method: 'POST',
+        apiServiceName: 'unlinkProject',
+    },
+    getUserDetails: {
+        path: '/user',
+        method: 'GET',
+        apiServiceName: 'getUserDetails',
+    },
+    createSubscription: {
+        path: '/subscriptions',
+        method: 'POST',
+        apiServiceName: 'createSubscription',
+    },
+    getSubscriptionList: {
+        path: '/subscriptions',
+        method: 'GET',
+        apiServiceName: 'getSubscriptionList',
+    },
+    deleteSubscription: {
+        path: '/subscriptions',
+        method: 'DELETE',
+        apiServiceName: 'deleteSubscription',
+    },
+    getSubscriptionFilters: {
+        path: '/subscriptions/filters',
+        method: 'POST',
+        apiServiceName: 'getSubscriptionFilters',
+    },
+};
+
+export const mattermostApiServiceConfigs: Record<MattermostApiServiceName, MattermostApiService> = {
+    getChannels: {
+        path: (args) => `/users/${args[0]}/teams/${args[1]}/channels`,
```

Comment:
@avas27JTG FYI this can accessed with the `@mattermost/client` library

Reply:
@mickmister Could you please clarify if you are suggesting to use `@mattermost/client` library for calling the API itself or for just getting the API endpoint to be used here in `path`?

also are you referring to this library https://www.npmjs.com/package/mattermost-client ?

Reply:
Used this now